server/api/accounts.js

```diff
@@ -16,15 +16,24 @@ const register = function (server, serverOptions) {
         path: '/api/accounts',
         options: {
             tags: ['api','accounts'],
+            description: 'Get a paginated list of all accounts (need admin role)',
+            notes: 'This endpoint returns a paginated list of accounts that have been created. This is a much longer detailed description if we have answers to common questions, but I mainly have questions. Whats the difference between accounts and users? When would you want to list all accounts instead of all users? Assuming you need an admin role to use this endpoint? Where is that configured?',
             auth: {
                 scope: 'admin'
             },
             validate: {
                 query: {
-                    sort: Joi.string().default('_id'),
-                    limit: Joi.number().default(20),
-                    page: Joi.number().default(1)
+                    sort: Joi.string().default('_id').description('This is a description for this query parameter. This is obviously a sort but what are the allowed values in the default setup? I\'ve added an example allow() validation which will show up as a pick list in interactive docs.').allow('_id', '_option2', '_option3'),
+                    limit: Joi.number().integer().default(20).description('I\'m assuming this is how many results you want at a time per page? We should mention that just so it\'s clear. It should also have min/max Joi validation which also shows up in docs. I\'ve added an example.').min(1).max(1000),
```

Comment:
How does `.number().integer()` work? Is this still valid Joi? Are both definitions needed?

Reply:
I've recently had to go neck deep into Joi validation so got caught up with this nuance stuff in the way Joi does things though it looks a little strange. number() validates as any type of number including a float. So asking for page 1.534 would pass validation though it shouldn't. .number().integer() validates as a round number. Not sure why they decided you had to validate as a number before integer but I assume there is a method to their madness. It's valid Joi and even on the example code at https://github.com/hapijs/joi